src/tiled/mapdocument.cpp

```diff
@@ -139,7 +139,7 @@ void MapDocument::saveSelectedObject(const QString &name, int groupIndex)
 
     if (ObjectTemplate *objectTemplate = model->saveObjectToDocument(object, name, groupIndex)) {
         // Convert the saved object into an instance and clear the changed properties flags
-        object->setTemplateRef({objectTemplate->templateGroup(), objectTemplate->id()});
+        object->setTemplateRef(objectTemplate->templateRef());
         object->setChangedProperties(0);
```

Comment:
Actually these lines should be covered by an undo command, right? Now that you've written `ReplaceObjectsWithTemplate`, you should probably use it here.

Reply:
Hmm, currently this would be a partial undo, as removing from the template group is not implemented. But at least it would reset the object to being a normal object.

Reply:
Yeah, I think partial undo would be fine here.